app/Console/Commands/video_data.php

```diff
@@ -0,0 +1,150 @@
+	 * @return void
+	 */
+	public function __construct(PhotoFunctions $photoFunctions, Extractor $metadataExtractor)
+	{
+		parent::__construct();
+
+		$this->photoFunctions = $photoFunctions;
+		$this->metadataExtractor = $metadataExtractor;
+	}
+
+	/**
+	 * Execute the console command.
+	 *
+	 * @return mixed
+	 */
+	public function handle()
+	{
+		set_time_limit($this->argument('timeout'));
+
+		$this->line(
+			sprintf(
+				'Will attempt to generate up to %s video thumbnails/metadata with a timeout of %d seconds...',
+				$this->argument('count'),
+				$this->argument('timeout')
+			)
+		);
+
+		$photos = Photo::where('type', 'like', 'video/%')
+			->where('width', '=', 0)
+			->take($this->argument('count'))
+			->get()
+		;
+
+		if (count($photos) == 0) {
+			$this->line('No videos require processing');
+			return 0;
+		}
+
+		foreach ($photos as $photo) {
+			$this->line('Processing '.$photo->title.'...');
+			$url = Config::get('defines.dirs.LYCHEE_UPLOADS_BIG').$photo->url;
+
+			if ($photo->thumbUrl != '') {
+				$thumb = Config::get('defines.dirs.LYCHEE_UPLOADS_THUMB').$photo->thumbUrl;
+				if (file_exists($thumb)) {
+					$urlBase = explode('.', $photo->url);
+					$thumbBase = explode('.', $photo->thumbUrl);
+					if ($urlBase !== $thumbBase) {
+						$photo->thumbUrl = $urlBase[0].'.'.$thumbBase[1];
+						rename($thumb, Config::get('defines.dirs.LYCHEE_UPLOADS_THUMB').$photo->thumbUrl);
+						$this->line('Renamed thumb to match the video file');
+					}
+				}
+			}
+
+			if (file_exists($url)) {
+				$info = $this->metadataExtractor->extract($url, $photo->type);
+
+				if ($info['width']) {
+					$this->line('Extracted metadata');
+					$photo->width = $info['width'];
+				}
+				if ($info['height']) {
+					$photo->height = $info['height'];
+				}
+				if ($info['focal']) {
+					$photo->focal = $info['focal'];
+				}
+				if ($info['aperture']) {
+					$photo->aperture = $info['aperture'];
+				}
+				if ($info['takestamp']) {
+					$photo->takestamp = $info['takestamp'];
+				}
+				if ($info['latitude']) {
+					$photo->latitude = $info['latitude'];
+				}
+				if ($info['longitude']) {
+					$photo->longitude = $info['longitude'];
+				}
+
+				if ($photo->thumbUrl === '' || $photo->thumb2x === 0 || $photo->small === '' || $photo->small2x === '') {
+					$frame_tmp = '';
+					try {
+						$frame_tmp = $this->photoFunctions->extractVideoFrame($photo);
+					} catch (\Exception $exception) {
+						Logs::error(__METHOD__, __LINE__, $exception->getMessage());
+					}
+					if ($frame_tmp !== '') {
+						$this->line('Extracted video frame for thumbnails');
+						if ($photo->thumbUrl === '' || $photo->thumb2x === 0) {
+							if (!$this->photoFunctions->createThumb($photo, $frame_tmp)) {
+								Logs::error(__METHOD__, __LINE__, 'Could not create thumbnail for video');
+							}
+						}
+						if ($photo->small === '' || $photo->small2x === '') {
+							$this->photoFunctions->createSmallerImages($photo, $frame_tmp);
```

Comment:
doesn't this also creates mediums ?

Reply:
No, I modified `createSmallerImages()` not to generate the mediums if the frame argument is non-empty.